scripts/windows/install_msys2.bat

```diff
@@ -0,0 +1,66 @@
+@echo off
+:: Pass -v as first argument to enable debug echo
+:: Script to download and install Msys2 from %MSYS2_URL%
+:: Output is suitable for piping scripts
+:: ENV vars will be updated accordingly
+::      WIN_MSYS2_ROOT
+::      MSYS2_URL
+:: Usage: install_msys2 /? | [<install_path>]
+
+SETLOCAL ENABLEEXTENSIONS
+@IF "%~1"=="-v" shift /1 & echo on & echo
+
+IF NOT "%~1"=="/?" IF NOT "%~1"=="--help" GOTO:START
+@echo Download and install Msys2 from netcologne.dl.sourceforge.net
+@echo Usage: %0 /? ^| [^<install_path^>]
+@echo     Default ^<install_path^> is "C:\tools\msys64" or WIN_MSYS2_ROOT env var
+@echo     Define MSYS2_URL env var to use different download url
+@echo     Env vars required by build scripts will be dumped
+exit /b 2
+
+:START
+IF NOT "%~1"=="" SET "WIN_MSYS2_ROOT=%~1"
+IF "%WIN_MSYS2_ROOT%"=="" SET "WIN_MSYS2_ROOT=C:\tools\msys64"
+IF "%MSYS2_URL%"=="" SET "MSYS2_URL=https://netcologne.dl.sourceforge.net/project/msys2/Base/x86_64/msys2-base-x86_64-20190524.tar.xz"
+
+IF EXIST "%WIN_MSYS2_ROOT%\mingw64" call:log Ok && GOTO :MSYS2_INSTALLED
+@call:log Target directory %WIN_MSYS2_ROOT%
+
+IF EXIST "%WIN_MSYS2_ROOT%" rd /s/q "%WIN_MSYS2_ROOT%"
+SET "PSAdmin=powershell Start-Process -wait -WindowStyle hidden 'powershell.exe'"
```

Comment:
This 'Inception' command is to be able to prepare a self-elevating shell for executing stuff requiring admin rights